nodestream/project/plugin.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass
+from typing import List
+
+from ..file_io import LoadsFromYamlFile
+from ..pipeline.scope_config import ScopeConfig
+
+
+@dataclass
+class PluginScope(LoadsFromYamlFile):
```

Comment:
At first I wasn't sure the difference between this class and a `PipelineScope`. Its obvious when looking at the code. 

However, I think what tripped me up was the name here. What about `PluginConfiguration` Given the documentation, that seems to align with the purpose of the class. 

> A `PluginScope` represents a collection of configuration for a plugin.

Reply:
Yeah I was struggling for what to call this. Works for me! I'll update it.